dali/python/nvidia/dali/pipeline.py

```diff
@@ -990,3 +996,62 @@ def iter_setup(self):
         For example, one can use this function to feed the input
         data from NumPy arrays."""
         pass
+
+
+def _discriminate_args(**kwargs):
+    """Split args on those applicable to Pipeline constructor and the rest."""
+    fca = inspect.getfullargspec(Pipeline.__init__)  # Full Ctor Args
+    ctor_args = dict(filter(lambda x: x[0] in fca.args or x[0] in fca.kwonlyargs, kwargs.items()))
+    fn_args = dict(filter(lambda x: x[0] not in ctor_args.keys(), kwargs.items()))
+    return ctor_args, fn_args
+
+
+def pipeline(fn=None, **pipeline_kwargs):
```

Comment:
I think this name can be misleading because we have
`nvidia.dali.pipeline.Pipeline` (the pipeline class) and `nvidia.dali.pipeline.pipeline` the decorator.
How about renaming to `pipeline_def` to avoid confusion?

Reply:
Yes, I agree, `pipeline_def` or `def_pipeline`, or `def_pipe` or `make_pipeline`. Whatever to no have three different nvidia.dali.pipeline.[Pp]ipeline things.

Reply:
Personally, I'm fine with `pipeline`, because we do already have the `Pipeline` in the same package and place. So we do already:
```python
from nvidia.dali.pipeline import Pipeline

pipe = Pipeline(...)
```
Why not doing
```python
from nvidia.dali.pipeline import pipeline

@pipeline
def pipe():
   ...
```
Let's leave more sophisticated naming for the user:
```python
import nvidia.dali.pipeline.pipeline as pipeline_decorator
import nvidia.dali.pipeline.Pipeline as pipeline_object
```